src/titiler/xarray/titiler/xarray/factory.py

```diff
@@ -0,0 +1,364 @@
+"""TiTiler.xarray factory."""
+
+from typing import Callable, List, Literal, Optional, Type, Union
+
+import rasterio
+from attrs import define, field
+from fastapi import Body, Depends, Path, Query
+from geojson_pydantic.features import Feature, FeatureCollection
+from geojson_pydantic.geometries import MultiPolygon, Polygon
+from pydantic import Field
+from rio_tiler.constants import WGS84_CRS
+from rio_tiler.models import Info
+from starlette.responses import Response
+from typing_extensions import Annotated
+
+from titiler.core.dependencies import (
+    CoordCRSParams,
+    CRSParams,
+    DatasetPathParams,
+    DefaultDependency,
+    DstCRSParams,
+    HistogramParams,
+    StatisticsParams,
+)
+from titiler.core.factory import TilerFactory as BaseTilerFactory
+from titiler.core.factory import img_endpoint_params
+from titiler.core.models.responses import InfoGeoJSON, StatisticsGeoJSON
+from titiler.core.resources.enums import ImageType
+from titiler.core.resources.responses import GeoJSONResponse, JSONResponse
+from titiler.core.utils import render_image
+from titiler.xarray.dependencies import (
+    DatasetParams,
+    PartFeatureParams,
+    TileParams,
+    XarrayIOParams,
+    XarrayParams,
+)
+from titiler.xarray.io import Reader
+
+
+@define(kw_only=True)
+class TilerFactory(BaseTilerFactory):
+    """Xarray Tiler Factory."""
+
+    reader: Type[Reader] = Reader
+
+    path_dependency: Callable[..., str] = DatasetPathParams
+
+    reader_dependency: Type[DefaultDependency] = XarrayParams
+
+    # Indexes/Expression Dependencies (Not layer dependencies for Xarray)
+    layer_dependency: Type[DefaultDependency] = DefaultDependency
+
+    # Dataset Options (nodata, reproject)
+    dataset_dependency: Type[DefaultDependency] = DatasetParams
+
+    # Tile/Tilejson/WMTS Dependencies (multiscale option)
+    tile_dependency: Type[TileParams] = TileParams
+
+    # Statistics/Histogram Dependencies
+    stats_dependency: Type[DefaultDependency] = StatisticsParams
+    histogram_dependency: Type[DefaultDependency] = HistogramParams
+
+    img_part_dependency: Type[DefaultDependency] = PartFeatureParams
+
+    # Custom dependency for /variable
+    io_dependency: Type[DefaultDependency] = XarrayIOParams
+
+    add_viewer: bool = True
+    add_part: bool = True
+
+    # remove some attribute from init
+    img_preview_dependency: Type[DefaultDependency] = field(init=False)
+    add_preview: bool = field(init=False)
```

Comment:
we don't want to have `preview` for now (I don't think any user would want this except if they control the dataset available)